aten/src/ATen/native/cuda/Indexing.cu

```diff
@@ -1674,55 +1622,21 @@ void index_select_out_cuda_impl(
 
```

Comment:
ah one more thing, couple lines above you'll see that index_select supports int32 indices, while gather doesn't.